File: src/main/java/cccs/oldversion/tag/DefaultTag.java
package cccs.oldversion.tag;

import org.apache.log4j.Logger;

import com.google.common.hash.BloomFilter;

import cccs.oldversion.RulesConf.RuleConf;

public class DefaultTag extends CachableTag {
    static Logger log = Logger.getLogger(DefaultTag.class.getName());

    public DefaultTag(String tagname, RuleConf ruleConf, BloomFilter<CharSequence> bloom) {
        super(tagname, ruleConf, bloom);
    }

    @Override
    public boolean evaluate() {
        log.debug("evaluating DefaultTag");
        pushOrPullFromCache();
        return currentValue();
    }

    @Override
    protected String makeBloomPutKey() {
        String prefix = "";
        if (ruleConf.groupby != null) {
            prefix = ruleConf.groupby.get(0);
        }
        return makeBloomKey(prefix, ruleConf.rulename, tagname, row);
    }

    @Override
    protected String makeBloomGetKey() {
        String prefix = "";
        if (ruleConf.groupby != null) {
            prefix = ruleConf.groupby.get(0);
        }
        return makeBloomKey(prefix, ruleConf.rulename, tagname, row);
    }
}

File: src/main/scala/cccs/fluxcapacitor/tag/DefaultTag.scala
package cccs.fluxcapacitor.tag

import org.apache.log4j.Logger

import com.google.common.hash.BloomFilter

import cccs.fluxcapacitor.RuleConf
import cccs.fluxcapacitor.TagCache

class DefaultTag(
    tagName: String,
    ruleConf: RuleConf,
    tagCache: TagCache
) extends CachableTag(tagName, ruleConf, tagCache) {

  override def evaluate(): Boolean = {
    log.debug("evaluating DefaultTag")
    pushOrPullFromCache()
    currentValue()
  }

  // by default uses the same get/put key values
  def makeBloomPutGetKey(): String = {
    if (ruleConf.groupby != null) {
      // if a groupby is specified
      val colname = ruleConf.groupby.head
      makeBloomKey(ruleConf.rulename, tagName, colname)
    } else {
      makeBloomKey(ruleConf.rulename, tagName)
    }
  }

  override protected def makeBloomPutKey() = makeBloomPutGetKey

  override protected def makeBloomGetKey() = makeBloomPutGetKey
}
